.github/workflows/workflow.yml

```diff
@@ -14,20 +17,51 @@ jobs:
           - ubuntu-latest
           - windows-latest
         ocaml-version:
-          - 4.09.1
+          - 4.04.2
+          - 4.05.0
+          - 4.06.1
+          - 4.07.1
           - 4.08.1
+          - 4.09.1
 
     runs-on: ${{ matrix.os }}
 
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
 
+      - name: Try to restore opam cache
+        if: runner.os != 'Windows'
+        id: opam-cache
+        uses: actions/cache@v2
+        with:
+          path: "~/.opam"
+          key: ${{ matrix.os }}-${{ matrix.ocaml-version }}
+
+      - name: Try to restore opam cache (Windows)
+        if: runner.os == 'Windows'
+        id: opam-cache-windows
+        uses: actions/cache@v2
+        with:
+          path: "/home/runneradmin/.opam"
+          key: ${{ matrix.os }}-a${{ matrix.ocaml-version }}
+
       - name: Use OCaml ${{ matrix.ocaml-version }}
         uses: avsm/setup-ocaml@v1
         with:
           ocaml-version: ${{ matrix.ocaml-version }}
 
+      - name: Get opam root path
+        id: opam
+        run: echo "$(opam var root)"
+
+      - name: Try to restore build cache
+        id: build-cache
+        uses: actions/cache@v2
+        with:
+          path: "_build"
+          key: ${{ matrix.os }}-${{ matrix.ocaml-version }}
```

Comment:
What does the "build cache" cache?

Reply:
It caches the build artifacts. Isn't really necessary I think!

Reply:
Indeed, let's remove this. It is not really necessary (the library builds very quickly anyway), it complicates matters, and the less state to carry around the better.

Reply:
Agree. It's not good to carry that state anyway.